tests/unit/Swift/Plugins/DefaultSenderPluginTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+class Swift_Plugins_DefaultSenderPluginTest extends \PHPUnit_Framework_TestCase
+{
+
+    public function testPluginChangesSenderEmail()
+    {
+        $message = Swift_Message::newInstance()
+            ->setSubject('...')
+            ->setBody('...')
+        ;
+
+        $plugin = new Swift_Plugins_DefaultSenderPlugin('john@example.com');
+
+        $evt = $this->_createSendEvent($message);
+
+        $this->assertEquals($message->getFrom(), array());
+
+        $plugin->beforeSendPerformed($evt);
+
+        $this->assertEquals($message->getFrom(), array('john@example.com' => ''));
+
+        $plugin->sendPerformed($evt);
+
+        $this->assertEquals($message->getFrom(), array());
+    }
+
+    public function testPluginChangesSenderEmailAndName()
+    {
+        $message = Swift_Message::newInstance()
+            ->setSubject('...')
+            ->setBody('...')
+        ;
+
+        $plugin = new Swift_Plugins_DefaultSenderPlugin('john@example.com', 'John Doe');
+
+        $evt = $this->_createSendEvent($message);
+
+        $this->assertEquals($message->getFrom(), array());
+
+        $plugin->beforeSendPerformed($evt);
+
+        $this->assertEquals($message->getFrom(), array('john@example.com' => 'John Doe'));
+
+        $plugin->sendPerformed($evt);
+
+        $this->assertEquals($message->getFrom(), array());
+    }
+
+    // -- Creation Methods
+
+    private function _createSendEvent(Swift_Mime_Message $message)
```

Comment:
Fixed. I used this file as a reference:
https://github.com/swiftmailer/swiftmailer/blob/master/tests/unit/Swift/Plugins/DecoratorPluginTest.php
which could mean there is some work to do in regard to code style in other files too.